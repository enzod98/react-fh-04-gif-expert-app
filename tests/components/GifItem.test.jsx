import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Pruebas en <GifItem />', () => { 

    const title = 'Goku ssj4'
    const url = 'https://enzodure.com/'

    test('debe de hacer match con el snapshot', () => {
        const {container } = render(<GifItem title={ title } url={ url } />);
        expect( container ).toMatchSnapshot();
    })

    test('Debe mostrar la imagen con el URL y el ALT indicado', () => {
        render( <GifItem title={ title } url={ url } /> )
        // screen.debug();
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( title );

    })

    test('Debe mostrar el título en un párrafo', () => {
        render( <GifItem title={ title } url={ url } /> )
        expect( screen.getByText( title ) ).toBeTruthy();   //Solo tiene en cuenta texto renderizado en el html, no sí está como valor de una propiedad

    })

 })