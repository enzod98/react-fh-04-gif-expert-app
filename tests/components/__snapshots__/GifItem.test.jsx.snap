// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Goku ssj4"
      src="https://enzodure.com/"
    />
    <p>
      Goku ssj4
    </p>
  </div>
</div>
`;
